doc/cephfs/disaster-recovery-experts.rst

```diff
@@ -316,3 +316,44 @@ operational.
     will contain holes in place of the missing data.
 
 .. _Symbolic link recovery: https://tracker.ceph.com/issues/46166
+
+
+Restoring file system after recovering monitor stores using OSDs
+----------------------------------------------------------------
+
+During rare occasions, all the monitor stores of a cluster may get corrupted.
+To recover the cluster in such a scenario, you need to rebuild the monitor
+stores using the OSDs, and get back the pools intact (active+clean state).
+However, the rebuilt monitor stores don't restore the FS/MDS maps. Additonal
+steps are required to bring back the file system. Currently, only the steps to
+recover a **single active MDS** file system have been identified, and tested.
+These are described below.
+
+Stop all the MDSs of the cluster.
+
+.. warning::
+
+     If the MDSs are not stopped, they may start wiping the pools when
+     recreating the file system.
```

Comment:
??

Reply:
Yeah, wiping pools was a lazy way of saying it. So if the MDSs are still up, while the `fs new` command is executed, I think the MDSs don't pick up the existing in-RADOS metadata and start overwriting the metadata. As mentioned here, https://github.com/ceph/ceph/blob/master/doc/cephfs/disaster-recovery-experts.rst#mds-map-reset
"...  is that doing a remove/new will leave rank 0 in 'creating' state, such that it would overwrite any existing root inode on disk and orphan any existing files. In contrast, the 'reset' command will leave rank 0 in 'active' state such that the next MDS daemon to claim the rank will go ahead and use the existing in-RADOS metadata."""